### PR TITLE
Next.js Migration

### LINKED ISSUE #47: Next.js Migration
Migrate CRA app to Next.js.  No new features, pure replication.

### PR DESCRIPTION
**Pull request by zachdt**
closes #47 

Notes for CI/CD
- NETWORK_URL is currently required as a environment variable for future wallet-connect support (this is a mainnet infura link)
- REACT_APP_NOTIFY_KEY has been renamed to NOTIFY_KEY
- Next.js apps use `next build` -> `next start` commands to deploy

Includes various improvements, including
- absolute imports
- server-side rendering for market pages
- static-site generation for future portfolio pages
- Next.js _app rendering for global CSS + layout (Page component)

I would not recommend deploying this version quite yet, there are more improvements to performance that need to be completed.  This app version, for better or for worse, is a nearly identical copy of the CRA app ported to Next.js, and includes unnecessary or non-functional code.

I still need to repair a slight style bug related to using _app.tsx to render global CSS (need to lock vertical scroll), so look out for a post-PR commit in this thread.

### CODE REVIEW

**Comment by zachdt** [comment]
G2G

File: app/src/components/Market/OptionsTable/OptionsTable.tsx [hunk 1]
```diff
@@ -1,21 +1,21 @@
...
+import { formatAddress } from '../../../utils'
 
-import Button from 'components/Button'
+import Button from '../../Button'
 import EmptyTable from '../EmptyTable'
-import LitContainer from 'components/LitContainer'
-import Table from 'components/Table'
-import TableBody from 'components/TableBody'
-import TableCell from 'components/TableCell'
-import TableRow from 'components/TableRow'
```
Comment (Alexangelj) [inline app/src/components/Market/OptionsTable/OptionsTable.tsx:18]: Ah I really liked doing 'components/', is there a way to do this in next?

File: app/src/components/Market/OrderCard/components/BuyOrMint/components/Buy.tsx [hunk 2]
```diff
@@ -2,16 +2,16 @@ import React, { useCallback, useState } from 'react'
...
-import Box from 'components/Box'
-import Button from 'components/Button'
-import Input from 'components/Input'
-import Label from 'components/Label'
-import Spacer from 'components/Spacer'
+import Box from '@/components/Box'
+import Button from '@/components/Button'
+import Input from '@/components/Input'
+import Label from '@/components/Label'
+import Spacer from '@/components/Spacer'
```
Comment (Alexangelj) [inline app/src/components/Market/OrderCard/components/BuyOrMint/components/Buy.tsx:9]: Oh I see, this is how next can make the imports global?

Comment (zachdt) [reply app/src/components/Market/OrderCard/components/BuyOrMint/components/Buy.tsx:9]: Yep thats how absolute imports work in next.js

File: app/src/constants/index.ts [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+export enum LocalStorageKeys {
+  Version = 'version',
+  DarkMode = 'darkMode',
+  Deadline = 'deadline',
+  Slippage = 'slippage',
+  Transactions = 'transactions',
+  Tokens = 'tokens',
+  TokenList = 'tokenList'
+}
```
Comment (Alexangelj) [inline app/src/constants/index.ts:70]: nice!

File: app/src/hooks/utils/useLocalStorage.ts [hunk 4]
```diff
@@ -0,0 +1,36 @@
+// Serialized implementation of LocalStorage hook from https://github.com/NoahZinsmeister/hypertext/blob/master/context.tsx
```
Comment (Alexangelj) [inline app/src/hooks/utils/useLocalStorage.ts:1]: naoh's stuff is great

File: app/src/components/Market/OrderCard/components/BuyOrMint/components/Buy.tsx (hunk 2 above)
Comment (zachdt) [reply app/src/components/Market/OrderCard/components/BuyOrMint/components/Buy.tsx:9]: I didn't replace all the absolute imports but whenever I came across them in a build error I replaced them.  In production I will only use them.

File: app/src/hooks/utils/useLocalStorage.ts (hunk 4 above)
Comment (zachdt) [reply app/src/hooks/utils/useLocalStorage.ts:1]: He writes suspiciously concise stuff very helpful

**Review by Alexangelj — APPROVED** [review]
LGTM!

File: app/src/hooks/utils/useLocalStorage.ts (hunk 4 above)
Comment (Alexangelj) [reply app/src/hooks/utils/useLocalStorage.ts:1]: hypertext is also beautiful

**Pull request merged**